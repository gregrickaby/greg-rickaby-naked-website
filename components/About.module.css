.about {
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 740px;
  padding: 0 1.2rem;
}

@media (min-width: 768px) {
  .about {
    padding: 0;
  }
}
