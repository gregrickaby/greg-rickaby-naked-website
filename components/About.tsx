import styles from "./About.module.css";

export default function About() {
  return (
    <div id="about" className={styles.about}>
      <h2>About</h2>
      <p>
        Way back in 1997, I built my first website on Geocities and have been
        passionate about the web since! In my spare time I enjoy{" "}
        <a href="https://gregrickaby.smugmug.com" rel="author">
          taking photos
        </a>
        , cheering for the{" "}
        <a href="https://www.packers.com/">Green Bay Packers</a>, and making
        pizzas.
      </p>

      <h3>Web Developer</h3>
      <p>
        I&apos;m a Tech Lead at{" "}
        <a href="https://americaneagle.com/">AmericanEagle.com</a>, an industry
        leader in web design, development, hosting, and digital marketing. I
        partner with clients and team members to build websites with Next.js and
        WordPress.
      </p>
      <p>
        View my r√©sum√© on{" "}
        <a href="https://www.linkedin.com/in/gregrickaby/" rel="author">
          LinkedIn
        </a>
        .
      </p>

      <h3>Published Author</h3>
      <p>
        In 2017, I wrote a children&apos;s book titled{" "}
        <a href="https://amzn.to/34QAxAQ">
          <em>
            <strong>Creating a Website for Dummies Jr</strong>
          </em>
        </a>{" "}
        which was published by{" "}
        <a href="https://www.wiley.com/en-us"> Wiley Global</a> under the{" "}
        <a href="https://www.dummies.com/">dummies‚Ñ¢Ô∏è</a> brand.
      </p>

      <p>
        The book guides young readers through the planning and development
        processes of building a website. It has been translated into other 2
        other languages, and currently has a 4.5-star rating on Amazon.
      </p>

      <p>
        Since being published, I&apos;ve also served as the Technical Editor on
        three other books:
      </p>
      <ul>
        <li>
          <a href="https://amzn.to/37XCflU" rel="nofollow">
            <em>WordPress for Dummies</em>
          </a>{" "}
          (Sabin-Wilson, 2021)
        </li>
        <li>
          <a href="https://amzn.to/2BgCg7i" rel="nofollow">
            <em>Professional WordPress Plugin Development</em>
          </a>{" "}
          (Williams et al., 2020)
        </li>
        <li>
          <a href="https://amzn.to/37MMDLp" rel="nofollow">
            <em>WordPress All-In-One For Dummies</em>
          </a>{" "}
          (Sabin-Wilson, 2019)
        </li>
      </ul>

      <h3>Photographer</h3>
      <p>
        In late 2019, I bought a mirrorless camera and spent months taking
        courses, reading, practicing, and having my work critiqued by peers.
        What has started as an enjoyable hobby, has turned into a passionate way
        to express my creativity.
      </p>
      <p>
        View my{" "}
        <a href="https://gregrickaby.smugmug.com" rel="author">
          portfolio
        </a>{" "}
        on SmugMug.
      </p>

      <h3>Contributions</h3>
      <p>
        I've been part of the WordPress community since 2008. I've made
        contributions to core, documentation, plugins, and themes. I've also
        spoken at WordCamps and meet-ups. I'm also a contributor to other
        open-source projects like Next.js, Storybook, and Gatsby.
      </p>
      <p>
        Outside of tech, I volunteer my time and efforts at local community
        organizations. The Boy Scouts, Wiregrass Church, and the Enterprise High
        School Band Boosters to name a few.
      </p>

      <p>
        You can find me around the web with the links below. Stay safe out
        there, and thanks for reading!
        <span role="img" aria-label="cheers">
          üçª
        </span>
      </p>
    </div>
  );
}
