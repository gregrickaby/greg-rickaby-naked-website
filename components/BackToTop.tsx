'use client'

import {IconArrowUp} from '@tabler/icons-react'
import {useEffect, useState} from 'react'

/**
 * The back to top component.
 */
export default function BackToTop() {
  const [showButton, setShowButton] = useState(false)
  const buttonText = 'Go back to the top of the page'

  /**
   * Scroll handler.
   */
  const handleScroll = () => {
    window.scrollTo({
      top: 0,
      behavior: 'smooth'
    })
  }

  /**
   * Effect for showing the back to top button.
   */
  useEffect(() => {
    // Scroll event handler.
    const handleScroll = () => {
      setShowButton(window.scrollY > 200)
    }

    // Add event listener.
    window.addEventListener('scroll', handleScroll)

    // Cleanup event listener.
    return () => window.removeEventListener('scroll', handleScroll)
  }, [])

  return showButton ? (
    <button
      aria-label={buttonText}
      className="button fixed bottom-8 right-3 rounded-full bg-gray-400 p-2 text-white shadow-lg hover:bg-gray-500 lg:right-12 dark:bg-gray-800 hover:dark:bg-gray-700"
      data-testid="back-to-top"
      onClick={handleScroll}
      title={buttonText}
    >
      <IconArrowUp height="32" width="32" />
      <span className="sr-only">{buttonText}</span>
    </button>
  ) : null
}
