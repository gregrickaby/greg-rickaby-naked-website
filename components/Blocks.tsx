'use client'

import '@fancyapps/ui/dist/fancybox/fancybox.css'
import {useEffect, useRef} from 'react'

/**
 * The blocks component.
 */
export default function Blocks({content}: Readonly<{content: string}>) {
  // Create a ref for the post content.
  const contentRef = useRef<HTMLDivElement>(null)

  /**
   * Initialize photo gallery.
   */
  async function galleryInit() {
    // On the server? Bail.
    if (typeof window === 'undefined') {
      return
    }

    // No ref? Bail.
    if (!contentRef.current) {
      return
    }

    // Attempt to find the gallery.
    const gallery = contentRef.current.querySelector('.grd-photo-gallery-grid')

    // No gallery? Bail.
    if (!gallery) {
      return
    }

    // Import the libraries on demand.
    const Masonry = (await import('masonry-layout')).default
    const Fancybox = (await import('@fancyapps/ui')).Fancybox
    const imagesLoaded = (await import('imagesloaded')).default

    // Initialize ImagesLoaded.
    const imagesLoadedInstance = imagesLoaded(gallery)

    // Initialize Masonry.
    const masonryGrid = new Masonry(gallery, {
      gutter: 24,
      itemSelector: '.grd-photo-gallery-image',
      columnWidth: '.grd-photo-gallery-grid-sizer',
      percentPosition: true
    })

    // When ImagesLoaded is done, initialize Masonry.
    imagesLoadedInstance.on('done', () => {
      masonryGrid?.layout?.()
    })

    // Initialize Fancybox.
    Fancybox.bind('[data-fancybox]', {
      animated: false,
      groupAll: true,
      defaultDisplay: 'flex',
      Toolbar: {
        display: {
          left: ['infobar'],
          middle: [
            'zoomIn',
            'zoomOut',
            'toggle1to1',
            'rotateCCW',
            'rotateCW',
            'flipX',
            'flipY'
          ],
          right: ['slideshow', 'thumbs', 'download', 'close']
        }
      }
    })

    // Destroy Masonry and Fancybox when the component unmounts.
    return () => {
      masonryGrid?.destroy?.()
      Fancybox.destroy?.()
    }
  }

  /**
   * Initialize the gallery when the component mounts.
   */
  useEffect(() => {
    galleryInit()
  }, [content])

  return (
    <div
      ref={contentRef}
      className="text-lg"
      dangerouslySetInnerHTML={{__html: content}}
    />
  )
}
