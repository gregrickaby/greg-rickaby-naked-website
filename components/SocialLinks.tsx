import config from '@/lib/config'
import React from 'react'

/**
 * Social links component.
 */
export default function SocialLinks() {
  return (
    <div className="flex flex-wrap justify-center gap-4 pb-4">
      {config.socials.map((social) => (
        <a
          aria-label={`follow on ${social.name}`}
          className="rounded bg-zinc-900 p-1 text-2xl text-white transition-colors duration-200 hover:bg-zinc-700"
          href={social.url}
          key={social.name}
          rel="author"
          title={`Follow on ${social.name}`}
        >
          {!!social.icon && React.createElement(social.icon)}
        </a>
      ))}
    </div>
  )
}
