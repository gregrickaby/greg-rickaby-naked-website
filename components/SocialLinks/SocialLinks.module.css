.socialLinks {
  @apply flex flex-wrap justify-center gap-4 pb-4;
}
