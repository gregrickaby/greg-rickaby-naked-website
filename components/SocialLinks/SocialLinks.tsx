import config from '@/lib/config'
import {createElement} from 'react'

/**
 * Social links component.
 */
export function SocialLinks() {
  return (
    <div
      className="flex flex-wrap justify-center gap-4 pb-4"
      data-testid="social-links"
    >
      {config.socials.map((social) => (
        <a
          aria-label={`follow on ${social.name}`}
          className="button"
          href={social.url}
          key={social.name}
          rel="author"
          title={`Follow on ${social.name}`}
        >
          {!!social.icon && createElement(social.icon)}
        </a>
      ))}
    </div>
  )
}
