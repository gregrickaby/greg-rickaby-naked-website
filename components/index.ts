export * from './BackToTop'
export * from './Blocks'
export * from './Footer'
export * from './Header'
export * from './JsonLd'
export * from './Navigation'
export * from './SocialLinks'
