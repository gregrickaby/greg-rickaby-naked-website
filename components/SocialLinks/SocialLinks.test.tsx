import {SocialLinks} from '@/components/SocialLinks'
import '@testing-library/jest-dom'
import {render} from '@testing-library/react'
import {axe} from 'jest-axe'
import {describe, expect, it} from 'vitest'

/**
 * Test suite for the SocialLinks component.
 */
describe('SocialLinks', () => {
  it('should render', () => {
    const {getByTestId} = render(<SocialLinks />)
    const socialLinks = getByTestId('social-links')
    expect(socialLinks).toBeInTheDocument()
  })

  it('should not have accessibility violations', async () => {
    const {container} = render(<SocialLinks />)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })

  it('should match snapshot', () => {
    const {getByTestId} = render(<SocialLinks />)
    const socialLinks = getByTestId('social-links')
    expect(socialLinks).toMatchSnapshot()
  })
})
