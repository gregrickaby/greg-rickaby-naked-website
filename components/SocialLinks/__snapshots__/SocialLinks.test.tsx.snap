// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SocialLinks > should match snapshot 1`] = `
<div
  class="_socialLinks_34e23a"
  data-testid="social-links"
>
  <a
    aria-label="follow on Instagram"
    class="button _socialButton_34e23a"
    href="https://www.instagram.com/gregoryrickaby/"
    rel="noopener noreferrer"
    title="Follow on Instagram"
  >
    <svg
      class="tabler-icon tabler-icon-brand-instagram "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 4m0 4a4 4 0 0 1 4 -4h8a4 4 0 0 1 4 4v8a4 4 0 0 1 -4 4h-8a4 4 0 0 1 -4 -4z"
      />
      <path
        d="M12 12m-3 0a3 3 0 1 0 6 0a3 3 0 1 0 -6 0"
      />
      <path
        d="M16.5 7.5l0 .01"
      />
    </svg>
  </a>
  <a
    aria-label="follow on Threads"
    class="button _socialButton_34e23a"
    href="https://www.threads.net/@gregoryrickaby"
    rel="noopener noreferrer"
    title="Follow on Threads"
  >
    <svg
      class="tabler-icon tabler-icon-brand-threads "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 7.5c-1.333 -3 -3.667 -4.5 -7 -4.5c-5 0 -8 2.5 -8 9s3.5 9 8 9s7 -3 7 -5s-1 -5 -7 -5c-2.5 0 -3 1.25 -3 2.5c0 1.5 1 2.5 2.5 2.5c2.5 0 3.5 -1.5 3.5 -5s-2 -4 -3 -4s-1.833 .333 -2.5 1"
      />
    </svg>
  </a>
  <a
    aria-label="follow on GitHub"
    class="button _socialButton_34e23a"
    href="https://github.com/gregrickaby/"
    rel="noopener noreferrer"
    title="Follow on GitHub"
  >
    <svg
      class="tabler-icon tabler-icon-brand-github "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 19c-4.3 1.4 -4.3 -2.5 -6 -3m12 5v-3.5c0 -1 .1 -1.4 -.5 -2c2.8 -.3 5.5 -1.4 5.5 -6a4.6 4.6 0 0 0 -1.3 -3.2a4.2 4.2 0 0 0 -.1 -3.2s-1.1 -.3 -3.5 1.3a12.3 12.3 0 0 0 -6.2 0c-2.4 -1.6 -3.5 -1.3 -3.5 -1.3a4.2 4.2 0 0 0 -.1 3.2a4.6 4.6 0 0 0 -1.3 3.2c0 4.6 2.7 5.7 5.5 6c-.6 .6 -.6 1.2 -.5 2v3.5"
      />
    </svg>
  </a>
  <a
    aria-label="follow on LinkedIn"
    class="button _socialButton_34e23a"
    href="https://linkedin.com/in/gregrickaby/"
    rel="noopener noreferrer"
    title="Follow on LinkedIn"
  >
    <svg
      class="tabler-icon tabler-icon-brand-linkedin "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 4m0 2a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v12a2 2 0 0 1 -2 2h-12a2 2 0 0 1 -2 -2z"
      />
      <path
        d="M8 11l0 5"
      />
      <path
        d="M8 8l0 .01"
      />
      <path
        d="M12 16l0 -5"
      />
      <path
        d="M16 16v-3a2 2 0 0 0 -4 0"
      />
    </svg>
  </a>
  <a
    aria-label="follow on WordPress"
    class="button _socialButton_34e23a"
    href="https://profiles.wordpress.org/gregrickaby/"
    rel="noopener noreferrer"
    title="Follow on WordPress"
  >
    <svg
      class="tabler-icon tabler-icon-brand-wordpress "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.5 9h3"
      />
      <path
        d="M4 9h2.5"
      />
      <path
        d="M11 9l3 11l4 -9"
      />
      <path
        d="M5.5 9l3.5 11l3 -7"
      />
      <path
        d="M18 11c.177 -.528 1 -1.364 1 -2.5c0 -1.78 -.776 -2.5 -1.875 -2.5c-.898 0 -1.125 .812 -1.125 1.429c0 1.83 2 2.058 2 3.571z"
      />
      <path
        d="M12 12m-9 0a9 9 0 1 0 18 0a9 9 0 1 0 -18 0"
      />
    </svg>
  </a>
  <a
    aria-label="follow on Amazon Author"
    class="button _socialButton_34e23a"
    href="https://www.amazon.com/author/gregrickaby"
    rel="noopener noreferrer"
    title="Follow on Amazon Author"
  >
    <svg
      class="tabler-icon tabler-icon-brand-amazon "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17 12.5a15.198 15.198 0 0 1 -7.37 1.44a14.62 14.62 0 0 1 -6.63 -2.94"
      />
      <path
        d="M19.5 15c.907 -1.411 1.451 -3.323 1.5 -5c-1.197 -.773 -2.577 -.935 -4 -1"
      />
    </svg>
  </a>
  <a
    aria-label="follow on Facebook"
    class="button _socialButton_34e23a"
    href="https://www.facebook.com/gregrickaby"
    rel="noopener noreferrer"
    title="Follow on Facebook"
  >
    <svg
      class="tabler-icon tabler-icon-brand-facebook "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 10v4h3v7h4v-7h3l1 -4h-4v-2a1 1 0 0 1 1 -1h3v-4h-3a5 5 0 0 0 -5 5v2h-3"
      />
    </svg>
  </a>
  <a
    aria-label="follow on Goodreads"
    class="button _socialButton_34e23a"
    href="https://www.goodreads.com/author/show/16999736.Greg_Rickaby"
    rel="noopener noreferrer"
    title="Follow on Goodreads"
  >
    <svg
      class="tabler-icon tabler-icon-book "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 19a9 9 0 0 1 9 0a9 9 0 0 1 9 0"
      />
      <path
        d="M3 6a9 9 0 0 1 9 0a9 9 0 0 1 9 0"
      />
      <path
        d="M3 6l0 13"
      />
      <path
        d="M12 6l0 13"
      />
      <path
        d="M21 6l0 13"
      />
    </svg>
  </a>
  <a
    aria-label="follow on CodePen"
    class="button _socialButton_34e23a"
    href="https://codepen.io/gregrickaby"
    rel="noopener noreferrer"
    title="Follow on CodePen"
  >
    <svg
      class="tabler-icon tabler-icon-brand-codepen "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 15l9 6l9 -6l-9 -6l-9 6"
      />
      <path
        d="M3 9l9 6l9 -6l-9 -6l-9 6"
      />
      <path
        d="M3 9l0 6"
      />
      <path
        d="M21 9l0 6"
      />
      <path
        d="M12 3l0 6"
      />
      <path
        d="M12 15l0 6"
      />
    </svg>
  </a>
  <a
    aria-label="follow on YouTube"
    class="button _socialButton_34e23a"
    href="https://www.youtube.com/@GregRickaby"
    rel="noopener noreferrer"
    title="Follow on YouTube"
  >
    <svg
      class="tabler-icon tabler-icon-brand-youtube "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2 8a4 4 0 0 1 4 -4h12a4 4 0 0 1 4 4v8a4 4 0 0 1 -4 4h-12a4 4 0 0 1 -4 -4v-8z"
      />
      <path
        d="M10 9l5 3l-5 3z"
      />
    </svg>
  </a>
  <a
    aria-label="follow on RSS Feed"
    class="button _socialButton_34e23a"
    href="https://gregrickaby.com/feed.xml"
    rel="noopener noreferrer"
    title="Follow on RSS Feed"
  >
    <svg
      class="tabler-icon tabler-icon-rss "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5 19m-1 0a1 1 0 1 0 2 0a1 1 0 1 0 -2 0"
      />
      <path
        d="M4 4a16 16 0 0 1 16 16"
      />
      <path
        d="M4 11a9 9 0 0 1 9 9"
      />
    </svg>
  </a>
  <a
    aria-label="follow on Feedly"
    class="button _socialButton_34e23a"
    href="https://feedly.com/i/subscription/feed%2Fhttps%3A%2F%2Fgregrickaby.com%2Ffeed.xml"
    rel="noopener noreferrer"
    title="Follow on Feedly"
  >
    <svg
      class="tabler-icon tabler-icon-brand-feedly "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.833 12.278l4.445 -4.445"
      />
      <path
        d="M10.055 14.5l2.223 -2.222"
      />
      <path
        d="M12.278 16.722l.555 -.555"
      />
      <path
        d="M19.828 14.828a4 4 0 0 0 0 -5.656l-5 -5a4 4 0 0 0 -5.656 0l-5 5a4 4 0 0 0 0 5.656l6.171 6.172h3.314l6.171 -6.172z"
      />
    </svg>
  </a>
</div>
`;
